src/foam/u2/PopupView.js

```diff
@@ -20,68 +20,77 @@ foam.CLASS({
+  ],
+
   axioms: [
     foam.u2.CSS.create({
       code: function() {/*
-      ^ {
-        background: #999;
-        box-shadow: 3px 3px 6px 0 gray;
-        color: white;
-        font-size: 18px;
-        opacity: 0.9;
-        padding: 20px;
-        position: absolute;
-        box-sizing: border-box;
-        z-index: 999;
-      }
+        ^ {
+          align-items: center;
+          bottom: 0;
+          display: flex;
+          justify-content: space-around;
+          left: 0;
+          position: fixed;
+          right: 0;
+          top: 0;
+          z-index: 1000;
+        }
+        ^container {
+          align-items: center;
+          display: flex;
+          height: 100%;
+          justify-content: space-around;
+          position: relative;
+          width: 100%;
+        }
+        ^background {
+          background-color: #000;
+          bottom: 0;
+          left: 0;
+          opacity: 0.4;
+          position: absolute;
+          right: 0;
+          top: 0;
+        }
+        ^inner {
+          z-index: 3;
+        }
       */}
     })
   ],
 
   properties: [
-    'x',
-    'y',
-    'width',
-    'height',
-    'maxWidth',
-    'maxHeight'
+    [ 'backgroundColor', '#fff' ]
   ],
 
   methods: [
-    function initE() {
-      var self     = this;
-      var parent   = this.parentNode;
-
-      if ( ! this.y       ) this.y = (parent.el().clientHeight - this.height)/2;
-      if ( ! this.x       ) this.x = (parent.el().clientWidth  - this.width )/2;
-      if ( this.width     ) this.style({width    : this.width     + 'px'});
-      if ( this.height    ) this.style({height   : this.height    + 'px'});
-      if ( this.maxWidth  ) this.style({maxWidth : this.maxWidth  + 'px'});
-      if ( this.maxHeight ) this.style({maxHeight: this.maxHeight + 'px'});
-
-      // Make a full-screen transparent background, which when clicked,
-      // closes this Popup
-      var bg = this.E('div').
-        style({
-          position: 'absolute',
-          width: '10000px',
-          height: '10000px',
-          opacity: 0,
-          top: 0,
-          zIndex: 998
-        }).
-        on('click', this.remove.bind(this)).
-        write();
-
-      this.
-        cssClass(this.myCls()).
-        style({
-          left: this.x + 'px',
-          top:  this.y + 'px'
-        }).
-        onunload.sub(bg.remove.bind(bg));
-
-      parent.style({position: 'relative'});
+    function init() {
+      this.SUPER();
+      var container = this.start().cssClass(this.myCls('container'));
+      container.start()
+          .cssClass(this.myCls('background'))
+          .on('click', this.close.bind(this))
+          .end();
+      var content = container.start()
+          .cssClass(this.myCls('inner'))
+          .style({ 'background-color': this.backgroundColor });
+      content.end();
+      container.end();
+      this.content = content;
+      this.cssClass(this.myCls());
+    },
+    function open() {
```

Comment:
This version doesn't have a notional parent. But since it covers the whole viewport, it's effectively modal. You have to close it to do anything else anyway.